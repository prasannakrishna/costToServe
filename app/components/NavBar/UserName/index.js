/**
 * Copyright Â© 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
 * <p>
 * This software is the confidential information of JDA Software, Inc., and is licensed
 * as restricted rights software. The use,reproduction, or disclosure of this software
 * is subject to restrictions set forth in your license agreement with JDA.
 */
// eslint-disable-next-line no-unused-vars
import React from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import { withStyles } from '@material-ui/core/styles';
import styled from '@emotion/styled';

const UserNameContainer = styled.div({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-around',
  padding: 2,
  '&::before': {
    content: '\'\'',
    display: 'block',
    marginTop: '8px',
    marginBottom: '8px',
    marginLeft: '8px',
    width: 12,
    paddingTop: 5,
    borderLeft: '1px solid rgba(255,255,255,0.5)',
  },
  '&::after': {
    cursor: 'pointer',
    fontFamily: 'Material Icons',
    content: 'expand_more',
    marginLeft: '-10%',
    marginTop: '5%',
    fontSize: '20px',
    color: 'rgba(255,255,255,0)',
  },
  '&:hover::after': {
    cursor: 'pointer',
    fontFamily: 'Material Icons',
    content: 'expand_more',
    marginLeft: '-10%',
    marginTop: '5%',
    fontSize: '20px',
    color: 'rgba(255,255,255,0.5)',
  },
});

const styles = () => ({
  icon: {
    width: 20,
    flexGrow: 1,
    margin: 7,
    color: 'white',
    height: 24,
  },
  userIcon: {
    width: 24,
    height: 24,
    backgroundColor: '#ffff',
    color: 'black',
    fontSize: 12,
  },
});
const getAvatarString = (nameStr) => {
    // check for @
  if (nameStr.indexOf('@') > -1 || nameStr.indexOf(' ') < 0) {
    return nameStr.split('@')[0].substring(0, 2).toUpperCase();
  }
  const y = nameStr.split(' ').map((e) => e.charAt(0));
  return `${y[0].toUpperCase()}${y[y.length - 1].toUpperCase()}`;
};
const UserName = (props) => {
  const { classes, userName } = props;
  if (!userName) return null;
  return (<UserNameContainer onClick={props.handleMenu}>
    <IconButton
      className={classes.icon}
    >
      <Avatar
        className={classes.userIcon}
      > {getAvatarString(userName)}
      </Avatar>
    </IconButton>
  </UserNameContainer>);
};

UserName.propTypes = {
  classes: PropTypes.object.isRequired,
  handleMenu: PropTypes.func,
  userName: PropTypes.string,
};

export default withStyles(styles)(UserName);
