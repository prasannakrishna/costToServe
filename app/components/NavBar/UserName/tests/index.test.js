/**
 * Copyright Â© 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
 * <p>
 * This software is the confidential information of JDA Software, Inc., and is licensed
 * as restricted rights software. The use,reproduction, or disclosure of this software
 * is subject to restrictions set forth in your license agreement with JDA.
 */
/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';

import UserName from '../index';

describe('<UserName />', () => {
  let handleMenu;
  let classes;
  let profile;
  let wrapper;
  beforeEach(() => {
    handleMenu = jest.fn();
    classes = {

    };
    profile = {
      data: {
        userName: 'User@User.user',
      },
    };
    wrapper = shallow(
      <UserName
        handleMenu={handleMenu}
        classes={classes}
        profile={profile}
      />
        ).dive();
  });
  describe('Expect to render component correctly', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

