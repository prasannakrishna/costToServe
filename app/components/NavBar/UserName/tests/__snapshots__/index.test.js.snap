// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserName /> Expect to render component correctly should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserName
    classes={
      Object {
        "icon": "UserName-icon-1",
        "userIcon": "UserName-userIcon-2",
      }
    }
    handleMenu={[MockFunction]}
    profile={
      Object {
        "data": Object {
          "userName": "User@User.user",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
